src/olympia/blocklist/mlbf.py

```diff
@@ -245,17 +245,42 @@ def generate_diffs(
         }
 
     def generate_and_write_stash(self, previous_mlbf: 'MLBF' = None):
-        # compare previous with current blocks
-        extras, deletes, _ = self.generate_diffs(previous_mlbf)[BlockType.BLOCKED]
+        """
+        Generate and write the stash file representing
+        changes between the previous and current MLBF Filters.
+
+        In order to support older FX clients that only understood blocked and unblocked
+        unblocked is a union of deletions from blocked and additions to soft_blocked.
+
+        Versions that move from hard to soft blocked will be picked up by old clients
+        as no longer hard blocked by being in the unblocked list.
+
+        Newer clients will also see soft blocked items as unblocked
+        but then will check the soft blocked list and re-add them to soft blocked.
```

Comment:
I am reading the PR top to bottom and this is one of the first comments... Q: is this adding complexity in the patch? If so, we can probably do without it because Product + others said we didn't need to account for hard to soft transitions.

Reply:
Well, it might not be the right question but that is still _my_ question, and a question that I want an answer for. That said, the code doesn't look too complicated so probably fine.

Reply:
I was trying to answer your question.. hope that I did. To quote the python style guide "simple is better than complex" but "complex is better than complicated" I think this patch is somewhere between simple and complex, but at least not complicated. All the complexity is in one place and changes elsewhere will not impact or be impacted by this code. Closing this thread.